Tests/DatabaseServiceTests/Repositories/UserRepositoryTests.cs

```diff
@@ -0,0 +1,213 @@
+                Email = "example@gmail.com",
+                PasswordHash = "hashpassword",
+                IsActive = true
+            };
+            #endregion
+        }
+
+        #region Create user tests
+        [Test]
+        public void CreateUserTest()
+        {
+            using var dbContext = new TPlatformDbContext(dbOptionsCreateUser);
+            repository = new UserRepository(mapper, dbContext);
+            repository.CreateUser(user, email);
+
+            DbUserComparer comparer = new DbUserComparer();
+            Assert.AreEqual(1, dbContext.Users.CountAsync().Result);
+            Assert.IsTrue(comparer.Equals(
+                dbUser,
+                dbContext
+                .Users
+                .FirstOrDefaultAsync()
+                .Result));
+        }
+
+        [Test]
+        public void CreateUserCredential()
+        {
+            using var dbContext = new TPlatformDbContext(dbOptionsCreateCredential);
+            repository = new UserRepository(mapper, dbContext);
+            repository.CreateUserCredential(credential);
+
+            DbUserCredentialsComparer comparer = new DbUserCredentialsComparer();
+            Assert.AreEqual(1, dbContext.UsersCredentials.CountAsync().Result);
+            Assert.IsTrue(comparer.Equals(
+                dbCredential,
+                dbContext
+                .UsersCredentials
+                .FirstOrDefaultAsync()
+                .Result));
+        }
+        #endregion
+
+        #region Delete user tests
+        [Test]
+        public void DeleteUserOk()
+        {
+            var options = new DbContextOptionsBuilder<TPlatformDbContext>()
+                .UseInMemoryDatabase(databaseName: "DeleteUserOkey")
+                .Options;
+
+            using var dbContext = new TPlatformDbContext(options);
+            repository = new UserRepository(mapper, dbContext);
+
+            dbContext.UsersCredentials.Add(dbUserCredential);
+            dbContext.SaveChanges();
+
+            repository.DeleteUser(userId);
+            var expectedField = dbContext.UsersCredentials.FirstOrDefault(uc => uc.UserId == userId).IsActive;
+
+            Assert.IsFalse(expectedField);
+        }
+
+        [Test]
+        public void DeleteUserIsActiveFalse()
+        {
+            var options = new DbContextOptionsBuilder<TPlatformDbContext>()
+                .UseInMemoryDatabase(databaseName: "DeleteUserIsActiveFalse")
+                .Options;
+
+            using var dbContext = new TPlatformDbContext(options);
+            repository = new UserRepository(mapper, dbContext);
+
+            dbContext.UsersCredentials.Add(dbUserCredential);
+            dbContext.SaveChanges();
+
+            dbUserCredential.IsActive = false;
+
+            var exception = Assert.Throws<BadRequestException>(() => repository.DeleteUser(userId));
+
+            Assert.AreEqual("User was deleted early or not confirmed", exception.Message);
+            // Come back
```

Comment:
what does it mean? Is this a reference to the terminator)?

Reply:
this tests was used SetUpAttribute, now we must change back instance object